api/proto/teleport/decision/v1alpha1/ssh_access.proto

```diff
@@ -159,6 +159,14 @@ message LockTarget {
 
   // ServerID is the host id of the Teleport instance.
   string server_id = 8;
+
+  // BotInstanceID is the bot instance ID if this is a bot identity; it is
+  // ignored otherwise.
+  string bot_instance_id = 9;
```

Comment:
If `bot_instance_id` and `join_token` can only exist together, should they be combined in a separate message?
What happens to the `user` field in case of a bot instance - is this worth a comment?

Reply:
It's a bit messy but it probably doesn't make sense to move these fields into a separate message. 

Bots are a variant of users and have a username (`bot-foo`, generally), so `user` will be set for both human users and bots. All bots have a `bot_instance_id`, but only some bots will have a `join_token` set. The fields will be empty for regular users, which is a convention shared with other fields, e.g. `server_id` is only set for servers, `mfa_device` is only set when a user has an MFA device, etc.